source/opt/dominator_analysis.cpp

```diff
@@ -55,12 +55,25 @@ bool DominatorAnalysisBase::Dominates(Instruction* a, Instruction* b) const {
     return tree_.Dominates(bb_a, bb_b);
   }
 
-  Instruction* current_inst = a;
-  while ((current_inst = current_inst->NextNode())) {
-    if (current_inst == b) {
+  const Instruction* current = a;
+  const Instruction* other = b;
+
+  if (tree_.IsPostDominator()) {
+    std::swap(current, other);
+  }
+
+  // We handle OpLabel instructions explicitly since they are not stored in the
+  // instruction list.
```

Comment:
@s-perron I wonder if labels should be stored in the instruction list?  I don't recall why they're not included.

Reply:
We did not make that decision.  It was that way when we join this project, and we did not change it.  It think it is separate just to make it easier to make sure that we do not accidentally get multiple OpLabel instructions per block and that it is at the start of the block.